clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2894,9 +2894,17 @@ bool CompilerInvocation::CreateFromArgs(CompilerInvocation &Res,
   if (LangOpts.OpenMPIsDevice)
     Res.getTargetOpts().HostTriple = Res.getFrontendOpts().AuxTriple;
 
-  // Set the triple of the host for SYCL device compile.
-  if (LangOpts.SYCLIsDevice)
+  if (LangOpts.SYCLIsDevice) {
+    // Set the triple of the host for SYCL device compile.
     Res.getTargetOpts().HostTriple = Res.getFrontendOpts().AuxTriple;
+    // If specified, create an empty integration header file for now.
+    const StringRef &HeaderName = LangOpts.SYCLIntHeader;
+    if (!HeaderName.empty()) {
+      int ResultFD = 0;
+      // Ignore the return value; compilation will fail if this file is absent.
+      (void)llvm::sys::fs::openFileForWrite(HeaderName, ResultFD);
```

Comment:
Do we care about doing something if this has failed?  

The 3rd param probably wants to be: CD_OpenAlways instead of the default.  Otherwise we end up truncating the file if it exists.

Finally, we definitely shouldn't be leaking this file handle, particularly this early in the process.

I'd suggest looking into openNativeFileForWrite (to get the write handle to pass to closeFile), or use https://llvm.org/doxygen/namespacellvm_1_1sys_1_1fs.html#ac90e775833cd4fd0da26c96bfa8f2f06.